src/picknik_ur_gazebo_config/objectives/sample_april_tag.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<root BTCPP_format="4" main_tree_to_execute="Sample April Tag">
+  <!-- ////////// -->
+  <BehaviorTree ID="Sample April Tag" _description="Samples the pose of an April tag and returns the average measurement." _favorite="true">
+      <Decorator ID="Repeat" num_cycles="{num_cycles}" name="SamplePose">
+        <Control ID="Sequence">
+        <Action ID="LoadObjectiveParameters" config_file_name="{apriltag_config}" parameters="{parameters}"/>
+        <Action ID="GetCameraInfo" topic_name="/wrist_mounted_camera/color/camera_info" message_out="{camera_info}"/>         
+        <Decorator ID="RetryUntilSuccessful" num_attempts="5">
+          <Control ID="Sequence">                      
+            <Action ID="GetImage" topic_name="/wrist_mounted_camera/color/image_raw" message_out="{image}"/>
+            <Action ID="DetectApriltags" image="{image}" camera_info="{camera_info}" parameters="{parameters}" detections="{detections}"/> 
+            <Action ID="GetDetectionPose" detections="{detections}" target_id="{tag_id}" target_label="" detection_pose="{detection_pose}"/>
+          </Control>
+        </Decorator>
+        <Action ID="AveragePoseStamped" pose_sample="{detection_pose}" avg_pose="{avg_pose}" max_distance="{max_distance}" max_rotation="{max_rotation}"/>
+        <!-- TODO: update this objective to allow rejecting noisy samples but fail after X attempts
+        <Fallback name="Average pose or sample more">
+            <Action ID="AveragePoseStamped"/>
+            <Control ID="Sequence">
+              <Script code="num_cycles={num_cycles - 1}" />
+              <Script code="num_failures={num_failures + 1}" />
+            </Control>
+        </Fallback> -->
```

Comment:
Is this TODO planned for this PR, or a later one?

Reply:
No not planned but would make the objective more robust. I assumed you are refactoring and re-writing this functionality so I'd be happy to remove the comment if we don't want it.